mnefun/_mnefun.py

```diff
@@ -1049,9 +1063,21 @@ def run_sss_command(fname_in, options, fname_out, host='kasga', port=22,
     remote_pos = op.join(work_dir, 'temp_%s_raw_sss.pos' % t0)
     print('%sOn %s: copying' % (prefix, host), end='')
     fname_in = op.realpath(fname_in)  # in case it's a symlink
-    cmd = ['scp', '-P' + port, fname_in, host + ':' + remote_in]
+    cmd = ['scp', '-P ' + port, fname_in, host + ':' + remote_in]
     run_subprocess(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-
+    
+    if fname_bads is not None:
+        if kwargs is not None:
+            if 'badlimit' in kwargs.keys():
+                options += ' -badlimit %d' % kwargs['badlimit']
+        options += ' -autobad on -skip 0 30'
+        cmd = ['ssh', '-p', port, host,
+               'maxfilter -f ' + remote_in + ' -o ' + remote_out + ' ' +
+               options]
+        with open(fname_bads, "w+") as f:
+            run_subprocess(cmd, stdout=f, stderr=subprocess.PIPE)
+        options = '-v -force'
```

Comment:
I don't think this is what we want to do. I think `run_sss_command` should always:

1. `rsync` to push to remote
2. run one `maxfilter` call remotely
3. retrieve any files or outputs
4. clean up the remote (remove the file)

So if in their analysis script a user has both `mf_autobad=True` and `movecomp=True`, it means you will go through this cycle twice (independently).

If you're worried about efficiency, we can make step (4) optional with a `remove_transferred=True`. If the user has `p.movecomp` then in the "mf_autobad" step we can call `remove_transferred=False`. And when movecomp is done it won't need to re-transfer.

Reply:
> 1. rsync to push to remote
> 2. run one maxfilter call remotely
> 3. retrieve any files or outputs

At the moment I am doing just one `maxfilter` call remotely and piping the `stdout` to a local file i.e., not writing to remote file and rsyncing, which is rather nice. The alternative requires piping the MF shell output to a remote file before pulling, and I couldn't figure out how to do do that with `run_subprocess`. 

> So if in their analysis script a user has both mf_autobad=True and movecomp=True, it means you will go through this cycle twice (independently).

Not exactly sure WYM here. At the moment the HP estimation step for movecomp is being handled by `run_sss_positions` so they are independent.

> 4.clean up the remote (remove the file)
At the moment the clean up is working just fine without any changes to `run_sss_command`.  Or do you mean something else?

> If the user has p.movecomp then in the "mf_autobad" step we can call remove_transferred=False. And when movecomp is being executed it won't need to re-transfer.

If I am understanding this correctly, we may have to modify 2-3 functions to track files with tmp names as per `time.time` to save a few minutes repeating file transfers.

Reply:
> At the moment I am doing just one maxfilter call remotely and piping the stdout to a local file i.e., not writing to remote file and rsyncing, which is rather nice. The alternative requires piping the MF shell output to a remote file before pulling, and I couldn't figure out how to do do that with run_subprocess.

Conceptually and code wise it makes things less clean to shoe-horn this extra `autobad` step in what is meant to be a generic `maxfilter` call-and-retrieve function.

By default the output of `run_subprocess` is the `stdout` of the process that was executed, so it should not be so difficult to get the output as a `str`:

https://github.com/mne-tools/mne-python/blob/06c7785aabe72fc7394cc6e6ba1c7d5aec1d2c76/mne/utils/misc.py#L85

I can push some commits to your branch to do this if you prefer?

> If I am understanding this correctly, we may have to modify 2-3 functions to track files with tmp names as per time.time to save a few minutes repeating file transfers.

Okay to keep things simpler, let's not worry about efficiency for now, and just do the `autobad` step in one call and the `headpos` in another.

Reply:
Sorry dude but I am still not sure WYM WRT to combining the `autobad` & `headpos` routines. Feel free to push to the branch.